src/shared/GitHub/GitHubAuthentication.cs

```diff
@@ -63,16 +66,24 @@ public async Task<AuthenticationPromptResult> GetAuthenticationAsync(Uri targetU
 
             if (modes == AuthenticationModes.None)
             {
-                throw new ArgumentException($"Must specify at least one {nameof(AuthenticationModes)}", nameof(modes));
+                throw new ArgumentException(@$"Must specify at least one {nameof(AuthenticationModes)}", nameof(modes));
             }
 
             if (TryFindHelperExecutablePath(out string helperPath))
             {
                 var promptArgs = new StringBuilder("prompt");
-                if ((modes & AuthenticationModes.Basic) != 0) promptArgs.Append(" --basic");
-                if ((modes & AuthenticationModes.OAuth) != 0) promptArgs.Append(" --oauth");
-                if (!GitHubHostProvider.IsGitHubDotCom(targetUri)) promptArgs.AppendFormat(" --enterprise-url {0}", targetUri);
-                if (!string.IsNullOrWhiteSpace(userName)) promptArgs.AppendFormat(" --username {0}", userName);
+                if (modes == AuthenticationModes.All)
+                {
+                    promptArgs.Append(" --all");
+                }
+                else
+                {
+                    if ((modes & AuthenticationModes.Basic) != 0) promptArgs.Append(" --basic");
+                    if ((modes & AuthenticationModes.OAuth) != 0) promptArgs.Append(" --oauth");
+                    if ((modes & AuthenticationModes.Pat)   != 0) promptArgs.Append(" --pat");
+                }
+                if (!GitHubHostProvider.IsGitHubDotCom(targetUri)) promptArgs.AppendFormat(" --enterprise-url \"{0}\"", targetUri);
```

Comment:
Do the targetUri and the userName fields need injection armoring??
I can't tell from the context here where the resulting string is going to go, but just wondering if it would be possible to attack things with a carefully crafted string...

Reply:
For example, if the userName field contained a double quote and then some special characters.

Reply:
So there _is_ an opportunity here to inject things to the args, for example if the username was literally:

```
alice" --pat "
```

..then you'd be starting the helper with the extra `--pat` argument as the expanded argument list would be:

```
.. --username "alice" --pat "" ..
```

However, there's no opportunity to execute another executable. This is because:

1. the executable being launched is discovered directly from the GCM Core installation directory,
2. the full executable path is specified as a completely separate argument to the .NET `ProcessStartInfo` class from the rest of the executable arguments,
3. .NET does the right thing, and these args are being passed eventually to CreateProcess (and not via a shell which would be vulnerable here to executing things it shouldn't).

So yes, it's possible inject something to the helper args, which isn't great as a creative username might mess up the prompt, but it doesn't present a threat.
The helpers are literally just there to show fancy input text boxes and buttons - they don't perform any logic themselves.

I'll look at adding better quoting and escaping of `"` however.

Reply:
Added escaping.